docs/key-issues/gpus.md

```diff
@@ -30,52 +30,66 @@ Packaging such projects for PyPI has been, and still is, quite challenging.
+no way to express this dependency. The same will be true for ROCm and other
+types of GPU hardware and languages.
+
+NVIDIA has made steps towards better support for CUDA on PyPI. Various
+library components of the CTK have been packaged as wheels and are now
+distributed on PyPI, such as
+[nvidia-cublas-cu11](https://pypi.org/project/nvidia-cublas-cu11/). Python
+wrappers around CUDA libraries have been consolidated into CUDA Python
+([website](https://developer.nvidia.com/cuda-python), 
+[PyPI package](https://pypi.org/project/cuda-python)), but this package assumes
+that the CUDA runtime and driver are already installed since it only provides
+Python bindings to the API. Many other projects remain hosted on
+[NVIDIA's Private PyPI Index](https://pypi.org/project/nvidia-pyindex/), which
+also includes rebuilds of TensorFlow and other packages.
 
 A single CUDA version supports a reasonable range of GPU architectures. New
 CUDA versions get released regularly, and - because they come with increased
 performance or new functionality - it may be necessary or desirable to build
 new wheels for that CUDA version. If only the supported CUDA version is
 different between two wheels, the wheel tags and filename will be identical.
 Hence it is not possible to upload more than one of those wheels under the same
-package name. To work around that, a project may either support only one CUDA
-version on PyPI, or create different packages. PyTorch and TensorFlow do the
-former, with TensorFlow supporting only a single CUDA version, and PyTorch
-providing more wheels for other CUDA versions and a CPU-only version in a
-separate wheelhouse (see
+package name.
+
+Historically, this required projects to produce packages specific CUDA minor
+versions. Projects would either support only one CUDA version on PyPI, or create
+different packages. PyTorch and TensorFlow do the former, with TensorFlow
+supporting only a single CUDA version, and PyTorch providing more wheels for
+other CUDA versions and a CPU-only version in a separate wheelhouse (see
 [pytorch.org/get-started](https://pytorch.org/get-started/locally/)).
 CuPy provides a number of packages: [`cupy`](https://pypi.org/project/cupy/),
 [`cupy-cuda102`](https://pypi.org/project/cupy-cuda102/),
 [`cupy-cuda110`](https://pypi.org/project/cupy-cuda1110/),
 [`cupy-cuda111`](https://pypi.org/project/cupy-cuda111/),
-[`cupy-cuda11x`](https://pypi.org/project/cupy-cuda11x/),
 [`cupy-rocm-4-3`](https://pypi.org/project/cupy-rocm-4-3/),
 [`cupy-rocm-5-0`](https://pypi.org/project/cupy-rocm-5-0/).
-Other projects do similar things - none of it works very well.
-
-GPU packages tend to result in very large wheels. This is true in particular
-for deep learning packages, because they link in
-[cuDNN](https://developer.nvidia.com/cudnn). For example, the most recent
-`manylinux2014` wheels for TensorFlow are 588 MB
+As of CUDA 11, CUDA promises [minor version
```

Comment:
IIRC the first version to actually achieve this was CUDA 11.2. Is that right? If so, it may be worth adding ".2"

Reply:
Technically it starts with 11.0, with the exception of NVRTC (which CuPy et al depends on) until 11.2. Also, minor ver compat in CUDA 11 does not cover PTX compilation. Not sure if theses technical details are important here. No strong opinion.

Reply:
Maybe as a footnote? Seems like the kind of thing that the majority of folks aren't interested in, but if you need it it's quite important and not easy to find.

Reply:
Hmm yeah this is a bit tricky. How deep into the weeds do we want to get here? If we mention PTX compilation, should we also mention that this is specifically an issue for `numba.cuda`, which provides additional plugins to get around this (the ptxcompiler and cubinlinker packages)? Do we want to mention the constraints for any other package that distributes PTX code (they need to restrict the PTX to instructions compiled for the oldest driver they want to use, which in practice means they probably need to compile on the oldest runtime they want to use, within a given major version)? What about how that applies to JIT outside of numba (for instance cupy has the ability to JIT, although I think in the long run there is interest in consolidating with numba, Leo could say more here)?

Reply:
Also the differences between CUDA 10 (which I imagine many stacks are still using somewhere), CUDA 11, and CUDA 12?

Reply:
If we could add the compat details and the PTX issue, with perhaps a link to a relevant Numba issue, in a footnote that would be really nice. It's the kind of info that is very difficult to find and hence valuable to have available.

CUDA 10 I'm less sure about, I'd be fine with leaving it out on the basis everyone is moving or has moved away from it.